service/src/main/java/org/broadinstitute/listener/relay/Utils.java

```diff
@@ -47,4 +50,28 @@ public static void writeCORSHeaders(
     responseHeaders.putIfAbsent(ACCESS_CONTROL_ALLOW_HEADERS, corsSupportProperties.allowHeaders());
     responseHeaders.putIfAbsent(ACCESS_CONTROL_MAX_AGE, corsSupportProperties.maxAge());
   }
+
+  public static boolean isValidOrigin(String origin, CorsSupportProperties corsSupportProperties) {
+    String stringToCheck;
+    // We want to strip the protocol.
```

Comment:
I'm going to go with since the ticket didn't specify schema, and the code base doesn't have an "allowed schemas" list, that we don't have to worry about it in this ticket.